src/prng.rs

```diff
@@ -120,39 +57,38 @@ pub enum PrngError {
 /// This type implements an iterator that generates a pseudorandom sequence of field elements. The
 /// sequence is derived from the key stream of AES-128 in CTR mode with a random IV.
 #[derive(Debug)]
-pub(crate) struct Prng<F: FieldElement, SC: StreamCipher> {
+pub(crate) struct Prng<F: FieldElement> {
     phantom: PhantomData<F>,
-    stream_cipher: SC,
+    expander: KeyExpander,
     buffer: Vec<u8>,
     buffer_index: usize,
     output_written: usize,
 }
 
-impl<F: FieldElement, SC: StreamCipher> Prng<F, SC> {
+impl<F: FieldElement> Prng<F> {
     /// Generates a seed and constructs an iterator over an infinite sequence of pseudorandom field
     /// elements.
-    pub(crate) fn new() -> Result<Self, PrngError> {
-        let mut seed = vec![0; SC::STREAM_CIPHER_SEED_SIZE];
-        getrandom(&mut seed)?;
-        Self::new_with_seed(&seed)
+    pub(crate) fn generate(suite: Suite) -> Result<Self, getrandom::Error> {
```

Comment:
Done, but I think this is a bit awkward because:
- `PrngError` propagates `SuiteError`;
- `VdafError` propagates `SuiteError`; and
- `VdafError` propagates `PrngError`.

I think our goal should be to do away with `PrngError` altogether and have the `prng` module propagate `SuiteError`. The only thing that's stopping us is that `extract_from_seed` needs to check that the input seed is the correct length. But once we do away with this internal API call (https://github.com/abetterinternet/libprio-rs/issues/78) we should be able to avoid this check.